functions/services/firebase/firebase.test.js

```diff
@@ -0,0 +1,309 @@
+require("jest-extended");
+const { MockFirebaseSdk, MockFirestore } = require("firebase-mock");
+
+function createMockSdk() {
+    const mockFirestore = new MockFirestore();
+    return new MockFirebaseSdk(
+        // use null if your code does not use RTDB
+        null,
+        // use null if your code does not use AUTHENTICATION
+        null,
+        // use null if your code does not use FIRESTORE
+        () => mockFirestore,
+        // use null if your code does not use STORAGE
+        null,
+        // use null if your code does not use MESSAGING
+        null
+    );
+}
+
+describe("Firebase", () => {
+    describe("#writeNewTeam", () => {
+        let writeNewTeam;
+        let mocksdk;
+
+        beforeEach(() => {
+            jest.resetModules();
+
+            mocksdk = createMockSdk();
+            jest.mock("firebase", () => mocksdk);
+
+            writeNewTeam = require("./").writeNewTeam;
+        });
+
+        afterEach(() => {
+            writeNewTeam = undefined;
+        });
+
+        it("should write new team when Firestore is empty", async () => {
+            let firestoreTeamsData;
+
+            let teamData = { OnFleetID: "<some-fleet-id>", zipcode: "342897" };
```

Comment:
I've recently added the faker library. So you could randomize the values like this:
```suggestion
            let teamData = { OnFleetID: faker.random.number(), zipcode: faker.address.zipCode() };
```

However, you must add `const faker = require('faker')` to the top.

There is a few places that you could utilize this :) and I'll leave it to you to utilize them where you see fit in the PR.

Reply:
Thank you, I will